src/client/components/pages/utils/absenceToVariant.ts

```diff
@@ -0,0 +1,19 @@
+import { ABSENCE_TYPE } from 'common/constants';
+import { FacultyAbsence } from 'common/dto/faculty/FacultyResponse.dto';
+import { TEXT_VARIANT } from 'mark-one';
+
+/**
+ * Finds the corresponding variant of type TEXT_VARIANT corresponding to
+ * the given absence
+ */
+export const absenceToVariant = (absence: FacultyAbsence): TEXT_VARIANT => {
+  if ([ABSENCE_TYPE.PARENTAL_LEAVE, ABSENCE_TYPE.RESEARCH_LEAVE]
+    .includes(absence.type)) {
+    return TEXT_VARIANT.NEGATIVE;
+  } if ([ABSENCE_TYPE.NO_LONGER_ACTIVE].includes(absence.type)) {
+    return TEXT_VARIANT.MEDIUM;
+  }
+  return TEXT_VARIANT.BASE;
```

Comment:
What do you think of this?

```suggestion
  switch (absence.type) {
    case ABSENCE_TYPE.PARENTAL_LEAVE:
    case ABSENCE_TYPE.RESEARCH_LEAVE:
      return TEXT_VARIANT.NEGATIVE;
    case ABSENCE_TYPE.NO_LONGER_ACTIVE:
      return TEXT_VARIANT.MEDIUM;
    default:
      return TEXT_VARIANT.BASE;
  }
```

Reply:
So:

1. This is just a suggestion :)
2. GitHub messed up/hid my comment on line 13 which said _"Should this be an `else if`?"_

Reply:
Originally I had `else if` and I got this linter error:
![image](https://user-images.githubusercontent.com/29589689/109322954-ea4af680-7820-11eb-99a5-1694fdd1c1d8.png)

Reply:
I updated this to use the switch statement - I think it's a little cleaner this way. Thanks!